astropy/visualization/time.py

```diff
@@ -0,0 +1,261 @@
+            else:
+                return super().__call__(value, pos=pos)
+
+        def format_ticks(self, values):
+            if len(values) == 0:
+                return []
+            if self._converter.format in YMDHMS_FORMATS:
+                times = Time(values, format='mjd', scale=self._converter.scale)
+                formatted = getattr(times, self._converter.format)
+                if self._converter.simplify:
+                    if self._converter.format in ('fits', 'iso', 'isot'):
+                        if all([x.endswith('00:00:00.000') for x in formatted]):
+                            split = ' ' if self._converter.format == 'iso' else 'T'
+                            formatted = [x.split(split)[0] for x in formatted]
+                    elif self._converter.format == 'yday':
+                        if all([x.endswith(':001:00:00:00.000') for x in formatted]):
+                            formatted = [x.split(':', 1)[0] for x in formatted]
+                return formatted
+            elif self._converter.format == 'byear_str':
+                return Time(values, format='byear', scale=self._converter.scale).byear_str
+            elif self._converter.format == 'jyear_str':
+                return Time(values, format='jyear', scale=self._converter.scale).jyear_str
+            else:
+                return super().format_ticks(values)
+
+
+    class MplTimeConverter(units.ConversionInterface):
+
+        def __init__(self, scale=None, format=None, simplify=None):
+
+            super().__init__()
+
+            self.format = format
+            self.scale = scale
+            self.simplify = simplify
+
+            # Keep track of original converter in case the context manager is
+            # used in a nested way.
+            self._original_converter = units.registry.get(Time)
+
+            units.registry[Time] = self
+
+        @property
+        def format(self):
+            return self._format
+
+        @format.setter
+        def format(self, value):
+            if value in UNSUPPORTED_FORMATS:
+                raise ValueError('time_support does not support format={0}'.format(value))
+            self._format = value
+
+        def __enter__(self):
+            return self
+
+        def __exit__(self, type, value, tb):
+            if self._original_converter is None:
+                del units.registry[Time]
+            else:
+                units.registry[Time] = self._original_converter
+
+        def default_units(self, x, axis):
+            if isinstance(x, tuple):
+                x = x[0]
+            if self.format is None:
+                self.format = x.format
+            if self.scale is None:
+                self.scale = x.scale
+            return 'astropy_time'
+
+        def convert(self, value, unit, axis):
+            """
+            Convert a Time value to a scalar or array.
+            """
+
+            # For Matplotlib < 2.2
```

Comment:
Should we increase our minimum version?

Reply:
I already just increased it to 2.1 from 2.0, but we could consider increasing further especially given the 2 year support for 4.0. I wish we had a written down policy for this :)

Reply:
Probably OK to leave for later: https://github.com/astropy/astropy/issues/8822#issuecomment-502203917